lib/rank_epitopes.py

```diff
@@ -0,0 +1,35 @@
+import pandas as pd
+
+class RankEpitopes:
+    def __init__(self, input_file, output_file):
+        self.input_file = input_file
+        self.output_file = output_file
+
+    def headers(self):
+        return [
+            'Gene Name',
+            'Mutation',
+            'Protein Position',
+            'HLA Allele',
+            'MT Epitope Seq',
+            'MT IC50',
+            'WT IC50',
+            'Fold Change',
+            'Tumor DNA Depth',
+            'Tumor DNA VAF',
+            'Tumor RNA Depth',
+            'Tumor RNA VAF',
+            'Gene Expression',
+            'Score',
+        ]
+
+    def execute(self):
+        df = pd.read_csv(self.input_file, sep='\t', index_col=False)
+        df['mt_score_rank'] = df['MT IC50'].rank(numeric_only=True, ascending=False, method='dense').fillna(value=0.0)
+        df['fold_change_rank'] = df['Fold Change'].rank(numeric_only=True, ascending=True, method='dense').fillna(value=0.0)
+        df['mt_allele_exp'] = df['Tumor RNA VAF'] * df['Gene Expression']
+        df['mt_allele_exp_rank'] = df['mt_allele_exp'].rank(numeric_only=True, ascending=True, method='dense').fillna(value=0.0)
+        df['tumor_dna_vaf_rank'] = df['Tumor DNA VAF'].rank(numeric_only=True, ascending=False, method='dense').fillna(value=0.0)
```

Comment:
Is this line right?  I expect that the above two lines ('mt_allele_exp' and 'Tumor DNA VAF') would both be ranked the same way, since higher scores are better for either.   The difference in "ascending" values makes me think that one of them isn't right.

Reply:
You are correct that this is the wrong order. I think I got Tumor DNA VAF mixed up with normal VAF in my head. Good catch. I will fix it tomorrow. 

Btw, `numeric_only=True` Means that NA values do not get ranked and the `fillna` Then sets NAs to 0.0. `method` defines how equal values get ranked. `dense` will assign them all the same rank and then ranks increase by 1 between groups. This seemed like the most reasonable option to me but you might want to check the documentation (https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.rank.html) to see if one of the other ones would be a better choice.